### PR TITLE
Implement m(atk) / m(def) percent system

### PR DESCRIPTION
**Pull request by skyleo**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Many skills use something called ATKPercentInfo & DEFPercentInfo in Aegis, the percentage bonuses are added up. (Both pre-re and re)

Every Bonus is in relation to it's skill and the strongest one according to abs(new_bonus) > abs(old_bonus) is chosen, when applying a new_bonus, if this is not met, then it won't be applied.

We're *slightly* deviating with the way we implement this, since Aegis keeps a map in order to keep track of the active Percent Buffs/Debuffs and their link to a skill, we won't do that.

Main purpose of this PR is to implement this system in order to be able to exclude these Percentage buffs from skills such as Asura Strike, which are not affected by Provoke or Curse for example.

**This introduces values to the enum scb_flag which requires it to be 64 bit. This will only work ~~MSVC 2022~~ with C23 or already in clang and gcc**

In order to improve MSVC nontheless, last commits show a solution by using an ifdefed typedef.

**Issues addressed:** #1532 

List of commits this is based on from my private fork, so I can compare if needed:
```
2309ba7d2efb30f043c1703aa69e58dc5c05279f
4a322ca9df8250f55af143382a8e95d46a0de084
c8f958f7ce842bb182d1918849a52ea81ee3a980
aa74964fb7248be46b6b342e3b8e1071abbcefca
5732261dc3642ac1cf058f9369787fc56b38a5ee
159d5167f9454ac4c1c8e34ae6e44bc90f201ec7
f94c2c68f62da10e72e10d70fa7be412204bbab7
bc744ff68f78e4c89c646d0dea00bcf24a961797
6d178d2188ebc1d706896d7b9f205f2169f2751f
69bebba647ffa8ca0a92dc901d18811a13c17196
35375883ee4edc15c21950041ab1ff0a95c7ee4d
f52ef302ad85266fc5989528142470b4e2ae0b46
5933b2c2f48c760e976d1f34bc1fe76978915c63
d3d33a560e01b88d2c14ba0cc5992f2d692ecf90
c2bf7a994767ad511fd37d6c9ab33d15259a3ea5
60b406dbb8c59d3b60a2ff5820f90abdcb7be544
907a3e15d95e652c2148d1cef5eb86478880be2d
8dfd03a0e3a8237725c9e2b98d83a6c69d96fff0
ce746187630b963af9aae377cb651ea40f13b5e2
0ee21c07a435f856ebb256fb93e371904058d762
6e619c64789691af1599a0c0aac1139882e052eb
b54ed44137e620a60cc333ec6d2cdc69efc0b9a3
d49ad9db026023eb2fec5526741576774e78b6e0
0865d2587f713454fb6e94a71a61becb3fa9851e
b11661a7794ec72476b16fbac4091d9fb85a11e0
a5ade750e3a977efa49a94dadde688203fb200b2
edaf3fd1882b8b80a8933b4003de2852598fa393
96cabeedc83ff57f4d5dc0708608f7e86722e425
9755566115e7a3cb40828fc869e3472d21b44706
05c812de9f5fab4b740c6b3f749ac22a511abb93
d044bf6629b3688e86bf2e5bde7bdeb5febbed9f
ad268787c070da7a4a82f8c15c03a5fa68505a99
```

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by skyleo** [comment]
Even VS 2022 apparently can't handle the enum no matter what settings for the compiler ...

Contemplating a solution atm that is least painful.

**Comment by skyleo** [comment]
Will leave the last 6 commits unsquashed for now and rebase after review.

**Review by guilherme-gm — COMMENTED** [review]
overall, code looks good to me. But I didn't test it nor compared to official. I plan to do that in a few days

File: src/map/status.c [hunk 1]
```diff
@@ -7740,7 +7741,7 @@ static int status_change_start_sub(struct block_list *src, struct block_list *bl
 					val2 = 25;
 				break;
 			case SC_NOEQUIPSHIELD:
-				if (!sd) //Def reduction
+				if (!sd) // DEF% reduction
```
[the remark this answers is not in the document]
Comment (skyleo) [reply src/map/status.c:7746]: Good catch, will do!

**Review by MishimaHaruna — APPROVED** [review]

**Review by MishimaHaruna — APPROVED** [review]

**Pull request merged**